bot/cogs/information.py

```diff
@@ -104,13 +104,35 @@ async def server_info(self, ctx: Context) -> None:
         member_count = ctx.guild.member_count
 
         # How many of each type of channel?
-        channels = Counter(c.type for c in ctx.guild.channels)
-        channel_counts = "".join(sorted(f"{str(ch).title()} channels: {channels[ch]}\n" for ch in channels)).strip()
+        channel_counter = Counter(c.type for c in ctx.guild.channels)
+        channel_type_list = []
+        for channel in channel_counter:
+            channel_type = str(channel).title()
+            channel_type_list.append(f"{channel_type} channels: {channel_counter[channel]}")
+
+        channel_type_list = sorted(channel_type_list)
+        channel_counts = "\n".join(channel_type_list).strip()
+
+        # How many channels are for staff only?
+        everyone_role = ctx.guild.roles[0]
+        hidden_channels = 0
+
+        for channel in ctx.guild.channels:
+            overwrites = channel.overwrites_for(everyone_role)
+            if overwrites.is_empty():
+                continue
+
+            for perm, value in overwrites:
+                if perm == 'read_messages' and value is False:
+                    hidden_channels += 1
```

Comment:
This will count `#:BIG RED CIRCLE: verification :BIG RED CIRCLE:` as a public channel, which it is, I guess; just not for actual members of our guild. I doubt anyone is going to do a manual count and compare the numbers, though.

On a more serious note: One reason this may fail in the future is if we decide to cherry-pick which channels we want to  showcase in Discovery. When changing the permissions over, I thought about that, but decided to postpone such a discussion. I think just leaving everything "lurkable" is fine.

Another thing is that Code Jam team channels will be displayed as "staff channels", since they won't be publicly visible.

Reply:
This is all true. But the only way to really properly address this would be to check if Helpers, Moderators or Admins had explicit read in these channels, I guess? Even that's not perfect.

Reply:
How about considering it a staff channel if the roles with read permissions are a subset of the roles with read permissions in the staff lounge? Maybe something like the event channel would get left out though.

Reply:
@SebastiaanZ this has now been addressed, see conversation below.